app/src/main/java/com/kickstarter/viewmodels/NewCardFragmentViewModel.kt

```diff
@@ -0,0 +1,206 @@
+package com.kickstarter.viewmodels
+
+import android.support.annotation.NonNull
+import com.kickstarter.R
+import com.kickstarter.libs.Environment
+import com.kickstarter.libs.FragmentViewModel
+import com.kickstarter.libs.rx.transformers.Transformers
+import com.kickstarter.libs.rx.transformers.Transformers.takeWhen
+import com.kickstarter.libs.rx.transformers.Transformers.values
+import com.kickstarter.services.ApolloClientType
+import com.kickstarter.ui.fragments.NewCardFragment
+import com.stripe.android.Stripe
+import com.stripe.android.TokenCallback
+import com.stripe.android.model.Card
+import com.stripe.android.model.Token
+import rx.Observable
+import rx.subjects.BehaviorSubject
+import rx.subjects.PublishSubject
+import type.PaymentTypes
+import java.lang.Exception
+
+interface NewCardFragmentViewModel {
+    interface Inputs {
+        /** Call when the card validity changes. */
+        fun card(card: Card?)
+
+        /** Call when the name field changes. */
+        fun name(name: String)
+
+        /** Call when the postal code field changes. */
+        fun postalCode(postalCode: String)
+
+        /** Call when the user clicks the save icon. */
+        fun saveCardClicked()
+
+        /** Call when the card input has focus. */
+        fun cardFocus(hasFocus: Boolean)
+    }
+
+    interface Outputs {
+        /** Emits when the drawable to be shown when the card widget has focus. */
+        fun cardWidgetFocusDrawable(): Observable<Int>
+
+        /** Emits when the password update was unsuccessful. */
+        fun error(): Observable<String>
+
+        /** Emits when the progress bar should be visible. */
+        fun progressBarIsVisible(): Observable<Boolean>
+
+        /** Emits when the save button should be enabled. */
+        fun saveButtonIsEnabled(): Observable<Boolean>
+
+        /** Emits when the card was saved successfully. */
+        fun success(): Observable<Void>
+
+    }
+
+    class ViewModel(@NonNull val environment: Environment) : FragmentViewModel<NewCardFragment>(environment), Inputs, Outputs {
+
+        private val card = PublishSubject.create<Card?>()
+        private val cardFocus = PublishSubject.create<Boolean>()
+        private val name = PublishSubject.create<String>()
+        private val postalCode = PublishSubject.create<String>()
+        private val saveCardClicked = PublishSubject.create<Void>()
+
+        private val cardWidgetFocusDrawable = BehaviorSubject.create<Int>()
+        private val error = BehaviorSubject.create<String>()
+        private val progressBarIsVisible = BehaviorSubject.create<Boolean>()
+        private val saveButtonIsEnabled = BehaviorSubject.create<Boolean>()
+        private val success = BehaviorSubject.create<Void>()
+
+        val inputs: NewCardFragmentViewModel.Inputs = this
```

Comment:
You can just do `val inputs: Inputs = this`

Reply:
Nice catch! Missed it when I moved it out of the Activity